### PR TITLE
gitea png to logo

### LINKED ISSUE #13969: Simpler Custom Logo Workflow
- Gitea version (or commit ref): develop, ee7133d135e1bf746ccd558371edd4fcdf185e7f

## Description

`generate-images.js` (which is an incredible piece of code!) creates logo pngs with 'gitea' in the title (ex. `gitea-lg.png`). I propose we use generic naming 'logo' (ex. `logo-lg.png`) to allow users to take advantage of the `generate-images` code and update the logo without having to customize the template file for each logo image reference.

Specifically, users would be able to replace `assets/logo.svg` with a custom logo and run `make generate-images` to have the logo updated across the app. Admittedly, we COULD use the code as is, but the logo name reference `gitea...` would technically not be correct.

I would add that this might be considered a breaking change, as it would result in many existing HTML template references to break. I'm not sure if the gitea community considers this when creating releases. 

I'm happy to implement a change if I can get some initial feedback.

Ex. code from `home.tmpl`
```
{{template "base/head" .}}
<div class="home">
	<div class="ui stackable middle very relaxed page grid">
		<div class="sixteen wide center aligned centered column">
			<div>
				<img class="logo" src="{{StaticUrlPrefix}}/img/gitea-lg.png" />
			</div>
			<div class="hero">
				<h1 class="ui icon header title">
					{{AppName}}
				</h1>
				<h2>{{.i18n.Tr "startpage.app_desc"}}</h2>
			</div>
		</div>
	</div>
```

### PR DESCRIPTION
**Pull request by kdumontnu**
Closes https://github.com/go-gitea/gitea/issues/13969

More general usage for logo png files to make it easier for users to customize logo.

Users should now be able to replace the file `logo.svg` with a custom svg logo and run `make generate-images` to update the logo (no need to change the templates). 

Note: I've kept `gitea.svg` and `gitea-192.png` for use in locations that should retain the gitea logo even after the fork logo is updated, such as the README files and webhook html templates. To regenerate these images, run `TAGS="gitea" make generate-images`.

Edit: After some discussion (see below), I've also updated most html template references to `logo.svg` instead of png.

### CODE REVIEW

**Comment by 6543** [comment]
It's kind breaking since lot of sites have customized there logo and have to change the path if merged ...

**Review by zeripath — APPROVED** [review]

**Review by lunny — CHANGES_REQUESTED** [review]
I don't think this is necessary. You can change the customized template to use your own logo.

**Comment by kdumontnu** [comment]
@6543 Yes, this is likely a breaking change

@lunny This simplifies the process and reduces the number of upstream conflicts in new forked projects. The idea here is twofold:
1. Users don't need to modify each template that references a logo file. They can now just replace the logo pngs or simply change `assets/logo.svg` and run `make generate-images`. 
2. It keeps a separate gitea logo file for places in template that should keep the gitea image even when the main logo is updated (ex. webhooks)

I'll leave it up to you and the rest of the maintainers whether a potentially breaking change for existing forks is worth it to reduce the number of breaking changes in future forks. I think so, but happy to close the PR if the team sees otherwise :+1:

**Review by techknowlogick — CHANGES_REQUESTED** [review]
Thanks for PR :)

Blocking until https://github.com/go-gitea/gitea/pull/13680 is merged due to potential merge conflicts that will arise from both PRs.

**Comment by kdumontnu** [comment]
> Thanks for PR :)
> 
> Blocking until #13680 is merged due to potential merge conflicts that will arise from both PRs.

K, I can rebase when that gets merged. 

One solution to avoid the breaking change is to keep the `gitea-X.png` files as they are, change the `generate-images` script and templates to `logo-X.png`, add `generate-images` to `make build`, but don't commit the `logo-X.png` files (not sure why we would commit these anyways). 

This way, anyone referencing `gitea-X.png` will still have access. It's a bit more involved change and would likely require a new CI test though.

**Comment by silverwind** [comment]
Rename is a good idea but I wonder if we can cut down the amount of different sizes. We could use SVG in HTML exclusively and it would be consistent with the favicon which we requires as SVG already.

It would get harder to use raster images for customization but apparently it's not impossible to embed them in SVG and with the favicon already changed, we have already committed to requiring a vector image.

**Comment by kdumontnu** [comment]
> Rename is a good idea but I wonder if we can cut down the amount of different sizes. We could use SVG in HTML exclusively and it would be consistent with the favicon which we requires as SVG already.
> 
> It would get harder to use raster images for customization but apparently it's not impossible to embed them in SVG and with the favicon already changed, we have already committed to requiring a vector image.

+1 to svg, I'll likely use svg in my fork. 

The only place `512` and `192` are referenced is `pwa/manifest_json.tmpl`, which I know very little about. Although apparently they [support svgs](https://html.spec.whatwg.org/multipage/semantics.html#attr-link-sizes) with `"sizes": "any"`.

**Comment by silverwind** [comment]
Full conversation will need more consideration and testing, I guess we can do it another time. I think the primary blockers are mobile browsers which use the PNGs for the homescreen icon. I can't find any definitive info on SVG support for those but I'd expect some issues.

For example Chrome devtools can not deal with `any`:

https://bugs.chromium.org/p/chromium/issues/detail?id=1107123

**Comment by kdumontnu** [comment]
> Full conversation to SVG will need more consideration and testing, I guess we can do it another time. I think the primary blockers are mobile browsers which use the PNGs for the homescreen icon. I can't find any definitive info on SVG support for those but I'd expect some issues.
> 
> For example Chrome devtools can not deal with `any`:
> 
> https://bugs.chromium.org/p/chromium/issues/detail?id=1107123

I can at least update all of the template html references to svg and leave the manifest alone for now. It resolves my concern and removes a future breaking change.

**Comment by silverwind** [comment]
Yeah, HTML is always fine to update.

**Comment by silverwind** [comment]
I think `fluid-icon` can also be removed if we can confirm that Firefox will use the SVG favicon.

https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/UI_considerations#How_an_icon_is_selected

**Review by techknowlogick — APPROVED** [review]
Dismissing my blocking review.

Thanks for this PR :)

File: templates/base/head.tmpl
Comment (lunny) [inline templates/base/head.tmpl]: Is this a missing one?

Comment (kdumontnu) [reply templates/base/head.tmpl]: Yes, just updated. Thanks.

**Comment by kdumontnu** [comment]
> Dismissing my blocking review.
> 
> Thanks for this PR :)

@techknowlogick You're very welcome. I've made some significant changes and rebased since you approved - can you take another look?

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
CI failed seems unrelated.

File: public/img/gitea.svg
Comment (silverwind) [inline public/img/gitea.svg]: Please output only minified SVGs to `public/img`.

File: Makefile
Comment (silverwind) [inline Makefile:702]: Please explain what this is doing in `docs/content/doc/developers/hacking-on-gitea.en-us.md`.

Comment (silverwind) [reply Makefile:702]: BTW that hacking doc is probably not the best place to talk about image customization, we should maybe move the paragraph to a page about customization.

Comment (jolheiser) [reply Makefile:702]: Hacking doc would be for anyone looking to change the logo as a contribution.

For someone who wants to change their _own_ logo for their instance, the instructions should be in https://docs.gitea.io/en-us/customizing-gitea/

Comment (silverwind) [reply Makefile:702]: Yes, `make generate-images` is meant for end users to build their custom logo, we should move the docs.

Comment (kdumontnu) [reply Makefile:702]: Should be all set. Could use some help with a Chinese translation.

Comment (silverwind) [reply Makefile:702]: I think chinese docs are already vastly out of date so I wouldn't worry about them. But sure, if someone's willing.

**Review by silverwind — APPROVED** [review]

**Comment by kdumontnu** [comment]
This PR should be all set to merge 👍

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
Thanks again :)

**Comment by silverwind** [comment]
As a next step, I think we should reduce the logos in use further, there is no reason to fetch `logo.svg` and `favicon.svg`, they are essentially the same resource. We can combine them into a 32px SVG `logo.svg`.

Also, we can possible cut down on the number of PNGs.

**Comment by kdumontnu** [comment]
Good read-up here: https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs

**Comment by silverwind** [comment]
Yeah, seen it. I disagree about the `.ico` thought, it's just unnecessary.

**Pull request merged**
